elasticsearch/client.go

```diff
@@ -101,6 +107,24 @@ func (cli *Client) post(ctx context.Context, index string, docType string, actio
 	return response, err
 }
 
+//CreateNewEmptyIndex is a method that creates an empty Elasticsearch index with the given indexName
+//It returns true if the index was created successfully, otherwise false
+func (cli *Client) CreateNewEmptyIndex(ctx context.Context, indexName string) (bool, error) {
+	esClient := dpelasticsearch.NewClient(cli.cfg.ElasticSearchAPIURL, cli.cfg.SignElasticsearchRequests, 5)
+	indexCreated := false
+	status, err := esClient.CreateIndex(ctx, indexName, GetSearchIndexSettings())
+	if err != nil {
+		log.Error(ctx, "error creating index. Make sure that ELASTIC_SEARCH_URL is set to http://localhost:11200", err)
+		return indexCreated, err
+	}
+	if status != http.StatusOK {
+		log.Error(ctx, "error creating index http status - "+strconv.Itoa(status), err)
```

Comment:
Our logging standards prohibit creating "dynamic messages". The message should always be the same for a given event. Any scenario specific context should be added via log data. Please can you update this to:

```
log.Error(ctx, "error creating elastic search index", log.Data{
    "response_status": status,
    "index_name": indexName
}, err)
```